services/mirror/mirror_pull.go

```diff
@@ -97,25 +105,28 @@ func parseRemoteUpdateOutput(output string) []*mirrorSyncResult {
 		refName := lines[i][idx+3:]
 
 		switch {
-		case strings.HasPrefix(lines[i], " * "): // New reference
-			if strings.HasPrefix(lines[i], " * [new tag]") {
-				refName = git.TagPrefix + refName
-			} else if strings.HasPrefix(lines[i], " * [new branch]") {
-				refName = git.BranchPrefix + refName
-			}
+		case strings.HasPrefix(lines[i], " * [new tag]"): // new tag
 			results = append(results, &mirrorSyncResult{
-				refName:     refName,
+				refName:     git.RefNameFromTag(refName),
+				oldCommitID: gitShortEmptySha,
+			})
+		case strings.HasPrefix(lines[i], " * [new branch]"): // new branch
+			refName = strings.TrimPrefix(strings.TrimSpace(refName), remoteName+"/")
+			results = append(results, &mirrorSyncResult{
+				refName:     git.RefNameFromBranch(refName),
 				oldCommitID: gitShortEmptySha,
 			})
 		case strings.HasPrefix(lines[i], " - "): // Delete reference
+			refName = strings.TrimPrefix(strings.TrimSpace(refName), remoteName+"/")
 			results = append(results, &mirrorSyncResult{
-				refName:     refName,
+				refName:     git.RefNameFromBranch(refName),
```

Comment:
What if delete a tag?

Reply:
good question. I have no idea.

Reply:
More context in #23070

I would suggest to complete the documents and tests by #23070 first

Otherwise nobody know what happens

Reply:
I think I have fixed this. The code is tricky. In a word,
`- [deleted] v1.0` should be a tag and `- [deleted] origin/branch` should be a branch.

Reply:
I have added unit tests and comments